src/functional/iterator/embedded.py

```diff
@@ -529,12 +553,22 @@ def _get_axes(
 
 
 def _make_tuple(
-    field_or_tuple: LocatedField | tuple, indices: int | slice | tuple[int | slice, ...]
-) -> tuple:  # arbitrary nesting of tuples of LocatedField
+    field_or_tuple: LocatedField | tuple,  # arbitrary nesting of tuples of LocatedField
+    indices: int | slice | tuple[int | slice, ...],
+    *,
+    as_column: bool = False,
+) -> tuple:  # arbitrary nesting of tuples of field values or `Column`s
     if isinstance(field_or_tuple, tuple):
         return tuple(_make_tuple(f, indices) for f in field_or_tuple)
     else:
-        return field_or_tuple[indices]
+        data = field_or_tuple[indices]
+        if as_column:
+            # wraps a vertical slice of an input field into a `Column`
+            return Column(
+                0, data
+            )  # TODO(havogt) when we support LocatedFields with origin (i.e. non-zero origin), kstart needs to be adapated here
```

Comment:
I tried to improve readability of this part by adding the `as_column` parameter and added comments. Also added 2 TODOs, but I wouldn't like to address them here.